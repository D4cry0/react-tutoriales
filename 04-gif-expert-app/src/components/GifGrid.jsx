// import { useState, useEffect } from "react";
import PropTypes from 'prop-types';

import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
//import { getGifs } from "../helpers/getGifs";

export const GifGrid = ({ category }) => {

    // esto reemplaza todo lo de abajo comentado
    const { images, isLoading} = useFetchGifs( category );

    /* // useEfect para disparar efectos secundarios
    // evitar llamadas a funciones de manera no eficiente
    // solo cuando es necesario

    const [images, setImages] = useState([]);

    // Se usan otras funciones porque el useEffect no permite retorno de promesas
    const getImages = async() => {
        const newImgs = await getGifs( category );
        setImages( newImgs );
    }

    // cada que se crea este componente, esta secciÃ³n se ejecuta
    // posterior a eso cuando se le llame y ya no lo hace
    // en cada momento que se redibuje algo al menos que sea necesario
    // explicitamente
    useEffect( () => {
        getImages();
    }, []); */


    return (
        <>
            <h3>{ category }</h3>
            {
                isLoading && ( <h2>Cargando...</h2> )
            }


            <div className="card-grid">
                { 
                    images.map( ( image ) => (
                        <GifGridItem 
                            key={ image.id }
                            { ...image } 
                        />
                    )) 
                }
            </div>
        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}