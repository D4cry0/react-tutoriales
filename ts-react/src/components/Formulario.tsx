import React from 'react'
import { useForm } from '../hooks/useForms'


interface FormData {
    email: string;
    nombre: string;
    edad: number;
}

export const Formulario = () => {
    
    const { formulario, handleChange } = useForm<FormData>({
        email: '',
        nombre: '',
        edad: 35
    });
    
    return (
        <form autoComplete='off'>
            <div className='mb-3'>
                <label className='form-label'>Email:</label>
                <input 
                    type='email'
                    className='form-control'
                    name='email'
                    value={ formulario.email }
                    onChange={ handleChange }
                />
            </div>
            <div className='mb-3'>
                <label className='form-label'>Nombre:</label>
                <input 
                    type='text'
                    className='form-control'
                    name='nombre'
                    value={ formulario.nombre }
                    onChange={ handleChange }
                />
            </div>

            { JSON.stringify( formulario ) }
        </form>

       
    )
}
