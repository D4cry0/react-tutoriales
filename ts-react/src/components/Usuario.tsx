import React, { useState } from 'react'


interface User {
    uid: string;
    name: string;
}

const temUser: User = {
    uid: 'xdf',
    name: 'juandddd'
}

export const Usuario = () => {

    const [user, setUser] = useState<User>({
        uid: '',
        name: ''
    });

    const login = () => {
        setUser({
            uid: 'ASFV3442',
            name: 'Arny C'
        })
    }

    

    return (
        <div className='mt-5'>
            <h3>Usuario: useState</h3>

            <button
                onClick={ login }
                className='btn btn-outline-primary'
            >
                Login
            </button>
            {
                (!user)
                    ? JSON.stringify(temUser)
                    : JSON.stringify(user)
            }
        </div>
    )
}
