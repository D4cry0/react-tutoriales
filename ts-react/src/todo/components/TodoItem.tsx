import React from 'react';
import { Todo } from '../interface/interfaces';
import { useTodos } from '../hooks/useTodos';


interface Props {
    todo: Todo;
}

export const TodoItem = ({ todo }: Props) => {

    // const { toggleTodo } = useContext( TodoContext );
    const { toggleTodo } = useTodos();

    return (
        <li 
            style={{
                cursor: 'pointer',
                textDecoration: todo.completed ? 'line-through' : ''
            }}
            onDoubleClick={ () => toggleTodo( todo.id ) }
        >
            { todo.desc }
        </li>
    )
}
