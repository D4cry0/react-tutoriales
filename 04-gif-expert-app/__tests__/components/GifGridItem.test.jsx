/**
 * @jest-environment jsdom
 */

import { render, screen } from "@testing-library/react";

import { GifGridItem } from "../../src/components/GifGridItem";

describe('Pruebas en <GifGridItem />', () => {

    const title = 'Maison GIF by Dreamcatcher';
    const url = 'https://media0.giphy.com/media/tVEhC7m5M6A99SZw9g/giphy.gif?cid=40ebb390eutzo6m0s9lyy10v79teg19m0pyldr1c9ygh8o6x&rid=giphy.gif&ct=g';

    test('Snapshot', () => {
        const { container } = render( <GifGridItem title={ title } url={ url } /> );

        expect( container ).toMatchSnapshot();
    });


    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render( <GifGridItem title={ title } url={ url } /> );
        
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );
        
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    });
    
    test('debe de mostrar el titulo en el componente', () => {
        render( <GifGridItem title={ title } url={ url } /> );
        
        expect( screen.getByText( title ) ).toBeTruthy();
    });
});