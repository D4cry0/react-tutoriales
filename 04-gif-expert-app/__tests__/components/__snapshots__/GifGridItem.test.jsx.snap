// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Maison GIF by Dreamcatcher"
      src="https://media0.giphy.com/media/tVEhC7m5M6A99SZw9g/giphy.gif?cid=40ebb390eutzo6m0s9lyy10v79teg19m0pyldr1c9ygh8o6x&rid=giphy.gif&ct=g"
    />
    <p>
      Maison GIF by Dreamcatcher
    </p>
  </div>
</div>
`;
